### PR TITLE
Redesign template components

### PR DESCRIPTION
**Pull request by dmorina**
-upgrade to ng 1.4.8 and angular-material 1.0.0
-redesign components

### CODE REVIEW

**Comment by mbernst** [comment]
:)

File: static/templates/template/components/text-edit.html [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    <md-input-container class="_short-input">
+      <label>Placeholder</label>
+      <input type="text" tabindex="0" ng-model="item.aux_attributes.placeholder"
+             aria-label="placeholder-input" placeholder="Answer here">
+    </md-input-container>
+    </div>
+  <div layout="row">
+    <md-input-container ng-if="item.sub_type!='number'" class="_short-input">
+      <label>Min character count</label>
```
Comment (aginzberg) [inline static/templates/template/components/text-edit.html]: these should probably have a min field

File: static/templates/template/components/text-edit.html [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+<md-input-container ng-if="item.sub_type=='text_area'" class="_text-input _long-text" md-no-float>
+  <textarea maxlength="{{item.aux_attributes.max_length}}" minlength="{{item.aux_attributes.min_length}}"
+            tabindex="0" ng-model="item.answer" ng-required="item.required"
+         aria-label="{{item.name}}" ng-disabled="true" placeholder="{{item.aux_attributes.placeholder}}"></textarea>
+</md-input-container>
+<div class="_item-properties" ng-if="item.isSelected" layout="column">
+  <ng-include src="'/static/templates/template/components/item_toolbar.html'"></ng-include>
+  <div layout="row">
+    <md-input-container class="_short-input">
+      <label>Type</label>
```
Comment (aginzberg) [inline static/templates/template/components/text-edit.html]: when i change this i lose focus

Comment (aginzberg) [inline static/templates/template/components/text-edit.html]: actually change anything for that matter.

Comment (dmorina) [inline static/templates/template/components/text-edit.html]: damn dropdowns :D they are not children of the div 
lemme have a look at it

File: static/js/template/controllers/template.controller.js [hunk 3]
```diff
@@ -226,11 +226,38 @@
...
+        }
+        function getTrustedUrl(url){
+            return $sce.trustAsResourceUrl(url);
+        }
+        function setDataSource(item, data_source){
+            if(!item.data_source || item.data_source != data_source){
+                item.data_source = data_source;
+                if(item.hasOwnProperty('value')) item.value = null;
+                if(item.hasOwnProperty('src')) item.src = null;
+                item.placeholder = 'will be filled from {' + data_source + '}';
```
Comment (aginzberg) [inline static/js/template/controllers/template.controller.js]: i really like this

File: static/templates/template/components/text-edit.html (hunk 2 above)
Comment (dmorina) [inline static/templates/template/components/text-edit.html]: fixed, dem scope apply's and improved the calculation

**Comment by aginzberg** [comment]
lgtm

**Pull request merged**